src/net.cpp

```diff
@@ -1131,15 +1131,17 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
         return;
     }
 
+    std::string strFrom = fLogIPs ? " from " + addr.ToString() : "";
```

Comment:
I see what you're doing here and understand, although it does seem a bit hacky.

I'm wondering if it would it be possible to instead set some class or instance-level (or maybe global) string to the "address" to be logged, and do something like you did above:

```
// you only do this once upon initialization, maybe close to where fLogIPs is set
std::string logMe = pfrom->GetId();
if (fLogIPs) {
  logMe = pfrom->addr.ToString();
}
...
```

.. then just log `logMe` (or whatever) every time instead of having conditions in the log lines and using the " from " trick.

Reply:
We don't have node id at this stage, it's assigned later (line 1183+). But I kind of applied your idea in other places. Force pushed.